### PR TITLE
INT-3784: Use acknowledge="transacted" by Default

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3784

Also
- suppress WARN log when priority mapping fails
- rework extract payload tests to use a single context

### CODE REVIEW

**Comment by garyrussell** [comment]
Repushed to fix the Log4 adjuster the same way.

File: spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsMessageDrivenEndpoint.java [hunk 1]
```diff
@@ -35,12 +36,33 @@
...
+	 * Construct an instance with an argument indicating whether the container's ack mode should
+	 * be overridden with {@link #setSessionAcknowledgeMode(String) sessionAcknowledgeMode}, default
+	 * 'transacted'.
+	 * @param listenerContainer the container.
+	 * @param listener the listener.
+	 * @param externalContainer true if the container is externally configured and should not have its ackmode
+	 * coerced when no sessionAcknowledgeMode was supplied.
+	 */
+	public JmsMessageDrivenEndpoint(AbstractMessageListenerContainer listenerContainer,
+			ChannelPublishingJmsMessageListener listener, boolean externalContainer) {
```
Comment (artembilan) [inline spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsMessageDrivenEndpoint.java]: This ctor looks strange from JavaConfig perspective, but what is the choice for XML...
I know! 
The `private` modification for this works well with the `BeanDefinition` :smile:

Comment (artembilan) [inline spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsMessageDrivenEndpoint.java]: That's only a concern.
Let me know and I'll fix it on merge
If rebase would be OK, of course...

Comment (garyrussell) [inline spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsMessageDrivenEndpoint.java]: Sounds good - yes this is for the XML - I recently changed the DSL to set transacted by default so it is fine.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/fd35d43aba7de42b67eb1ef74355f2507d6e5199

**Pull request closed without merging**